### ISSUE TITLE
Support for compressed MFEM-format mesh files?

### ISSUE THREAD

**Issue by rcarson3** [mfem/mfem#1540]
I've run into an interesting problem while prepping some larger production runs I'm planning on doing where I've got around a 27e6 element linear hex mesh. I've noticed that when saved this mesh is around 3 GBs in size using the MFEM mesh file format v1.0. So, I was wondering if there was a way to save this off in a compressed format that could later be read and then have some pre-processing done on it like suggested within #934 (also where that could be saved off in a compressed format) before I run these on a HPC system like Summit?

Alternatively, do you know if the ```CUBIT``` file format can be read in the parallel? I might be able to convert mesh over to that format with the parallel partitioning already done on the CUBIT side of things, and then load the mesh up that way in parallel. Although, I might run into some issues related to having to define the boundary element attributes over again.

**Comment by tzanio** [comment]
If you build with `MFEM_USE_ZLIB` you can just gzip e.g. the mesh file and read it with the `ifgzstream` classes. I believe this "just works" in example 1:

```
> gzip ../data/escher.mesh
> ./ex1 -m square-disc.mesh.gz
```

There is also compression in @pazner's ParaView output, though that's just for visualization.

**Comment by jandrej** [comment]
> Alternatively, do you know if the CUBIT file format can be read in the parallel? I might be able to convert mesh over to that format with the parallel partitioning already done on the CUBIT side of things, and then load the mesh up that way in parallel. Although, I might run into some issues related to having to define the boundary element attributes over again.

Right now it is not possible to read a pre-partitioned exodus mesh. But that was a good reminder to put that on the todo list.

**Comment by tzanio** [comment]
The outstanding item here is to create a miniapp that can partition a parallel mesh in `P` pieces and save the `P` files so they can compressed and read later in parallel. 

We can do that now if the miniapp runs on `P` MPI tasks, but it is desirable to be able to run the miniapp with less MPI tasks (and maybe even a single MPI task).

Anyone interested to take on this?

**Comment by tzanio** [comment]
The parallel mesh format read/write example is here: https://github.com/mfem/mfem/pull/1561/files#diff-acdeb0d83bfa7eabdf562ce5149e36fb

**Comment by tzanio** [comment]
This has been taken as in internal todo item.

**Comment by v-dobrev** [comment]
See https://github.com/mfem/mfem/pull/2669.
